### PR TITLE
Task Information date amendments

### PR DESCRIPTION
**Pull request by 247benji**
### Portal

- Edited TaskInformation property (DmEndDate, EffectiveReceiptDate, ExternalEndDate) data types to DateTime?
- Null check is performed in view to determine whether or not to display N/A
- Updated unit tests

### CODE REVIEW

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml [hunk 1]
```diff
@@ -15,11 +15,35 @@
...
+                                @{
+                                    if (Model.DmEndDate == null)
+                                    {
+                                        @:N/A
+                                    }
+                                    else
+                                    {
+                                        @Html.DisplayFor(model => model.DmEndDate)
+
+                                    }
```
Comment (SamirHassoun) [inline src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml:28]: You could do this on the TaskViewModel:
[DisplayFormat(NullDisplayText = "N/A")]
public DateTime? DmEndDate { get; set; }

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml [hunk 2]
```diff
@@ -15,11 +15,35 @@
...
+                                @{
+                                    if (Model.EffectiveReceiptDate == null)
+                                    {
+                                        @:N/A
+                                    }
+                                    else
+                                    {
+                                        @Html.DisplayFor(model => model.EffectiveReceiptDate)
+
+                                    }
```
Comment (SamirHassoun) [inline src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml:44]: You could do this on the TaskViewModel:
[DisplayFormat(NullDisplayText = "N/A")]
public DateTime? DmEndDate { get; set; }

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml [hunk 3]
```diff
@@ -29,7 +53,19 @@
...
+                                @{
+                                    if (Model.ExternalEndDate == null)
+                                    {
+                                        @:N/A
+                                    }
+                                    else
+                                    {
+                                        @Html.DisplayFor(model => model.ExternalEndDate)
+
+                                    }
```
Comment (SamirHassoun) [inline src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml:66]: You could do this on the TaskViewModel:
[DisplayFormat(NullDisplayText = "N/A")]
public DateTime? ExternalEndDate{ get; set; }

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml.cs [hunk 4]
```diff
@@ -33,17 +33,20 @@ public class _TaskInformationModel : PageModel
...
 
         [DisplayName("Effective Receipt Date:")]
-        public string EffectiveReceiptDate { get; set; }
+        [DisplayFormat(DataFormatString = "{0:d}")]
+        public DateTime? EffectiveReceiptDate { get; set; }
 
         [DisplayName("External End Date:")]
-        public string ExternalEndDate { get; set; }
+        [DisplayFormat(DataFormatString = "{0:d}")]
+        public DateTime? ExternalEndDate { get; set; }
```
Comment (SamirHassoun) [inline src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml.cs:49]: You can add [DisplayFormat(NullDisplayText = "N/A")] to DmEndDate, EffectiveReceiptDate, and ExternalEndDate

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml (hunk 1 above)
Comment (247benji) [reply src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml]: Cheers. Added NullDisplayText = "N/A" to DisplayFormat and kept @Html.DisplayFor(model => model.DmEndDate)

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml (hunk 3 above)
Comment (247benji) [reply src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml]: Cheers. Added NullDisplayText = "N/A" to DisplayFormat and kept @Html.DisplayFor(model => model.DmEndDate)

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml (hunk 2 above)
Comment (247benji) [reply src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml]: Cheers. Added NullDisplayText = "N/A" to DisplayFormat and kept @Html.DisplayFor(model => model.DmEndDate)

File: src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml.cs (hunk 4 above)
Comment (247benji) [reply src/Portal/Portal/Pages/DbAssessment/_TaskInformation.cshtml.cs]: Cheers. Added NullDisplayText = "N/A" to DisplayFormat.

**Pull request merged**